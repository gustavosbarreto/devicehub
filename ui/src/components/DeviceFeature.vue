<template>
<fragment>

  <div class="d-flex pa-0 align-center">
  <h1>Device Feature</h1>
  <v-spacer/>
  <v-spacer/>
  <!-- <AddDevice/> -->
  <!-- <v-btn outlined @click="$store.dispatch('modals/showAddDevice', true)">Add Device</v-btn> -->
  </div>
    <v-card class="mt-2">
        <v-app-bar flat color="transparent">
        </v-app-bar>
        <v-divider></v-divider>
        <v-card-text>
          
          <div class="item">
            <div class="item-name">Uid: </div>
            <div class="item-description">{{device.uid}}</div>
          </div>

          <div class="item">
            <div class="item-name">Name: </div>
            <!-- <v-icon small left>mdi-file-edit</v-icon>  -->
            <div class="item-description">{{device.name}}</div>
          </div>

          <div class="item">
            <div class="item-name">Mac: </div>
            <div class="item-description">{{device.identity}}</div>
          </div>

          <div class="item"> 
            <div class="item-name">Operating System: </div>
            <div class="item-description">{{device.attributes}}</div>
          </div>

          <div class="item">
            <div class="item-name">Public Key: </div>
            <div>{{device.public_key}}</div>
          </div>

          <div class="item">
            <div class="item-name">Tenant Id: </div>
            <div class="item-description">{{device.tenant_id}}</div>
          </div>

          <div class="item">
            <div class="item-name">Last Seen: </div>
            <div class="item-description">{{device.last_seen}}</div>
          </div>

          <div class="item">
            <div class="item-name">Online: </div>
            <div class="item-description">{{device.online}}</div>
          </div>

          <div class="item">
            <div class="item-name">Namespace: </div>
            <div class="item-description">{{device.namespace}}</div>
          </div>

        </v-card-text>
        <v-snackbar v-model="copySnack" :timeout=3000>Device SSHID copied to clipboard</v-snackbar>
    </v-card>
</fragment>
</template>

<script>
// import TerminalDialog from "@/components/TerminalDialog.vue";
// import AddDevice from "@/components/AddDevice.vue";

export default {
  name: "DeviceFeature",


  components: {
    // TerminalDialog,
    // AddDevice

  },

  created() {
      this.uid = this.$route.params.id
      this.$store.dispatch("devices/get", this.uid);
  },

  computed: {
    device() {
      return this.$store.getters["devices/get"];
    }
  },

  methods: {
    address(item) {
      return `${item.namespace}.${item.name}@${this.hostname}`;
    },

    copy(device) {
      this.$clipboard(device.uid);
    },

    remove(uid) {
      if (confirm("Are you sure?")) {
        this.$store.dispatch("devices/remove", uid);
      }
    },

    showCopySnack() {
      this.copySnack = true;
    },

    save(item) {
      this.$store.dispatch("devices/rename", {
        uid: item.uid,
        name: this.editName
      });

      item.name = this.editName;
    }
  },

  data() {
    return {
      uid: '',
      hostname: window.location.hostname,
      deviceIcon: {
        arch: "fl-archlinux",
        ubuntu: "fl-ubuntu"
      },
      copySnack: false,
      editName: "",
      headers: [
        // {
        //   text: "Online",
        //   value: "online",
        //   align: "center"
        // },
        {
          text: "Name",
          value: "name"
        },
        // {
        //   text: "Operating System",
        //   value: "attributes.pretty_name"
        // },

        // {
        //   text: "SSHID",
        //   value: "namespace",
        //   align: "center"
        // },
        // {
        //   text: "Actions",
        //   value: "actions",
        //   align: "center",
        //   sortable: false
        // }
      ]
    };
  }
};
</script>
<style scoped>
.merda {
  font-family: monospace;
}

.item {
  margin-bottom: 4px;
}
.item-name {
  font-size: 14px;
  color: #FFFFFF; 
  display:inline;
}

.item-description {
  font-size: 14px;
  display:inline;
}



</style>