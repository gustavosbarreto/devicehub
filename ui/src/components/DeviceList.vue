<template>
<fragment>

  <div class="d-flex pa-0 align-center">
  <h1>Devices</h1>
  <v-spacer/>
  <v-spacer/>
  <AddDevice/>
  <v-btn outlined @click="$store.dispatch('modals/showAddDevice', true)">Add Device</v-btn>
  </div>
    <v-card class="mt-2">
        <v-app-bar flat color="transparent">
        </v-app-bar>
        <v-divider></v-divider>
        <v-card-text class="pa-0">  <!-- @click:row="featureDevice" -->
            <v-data-table :headers="headers" :items="devices" item-key="uid" disable-pagination hide-default-footer>
                <template v-slot:item.online="{ item }">
                    <v-icon color="success" v-if="item.online">check_circle</v-icon>
                    <v-tooltip bottom v-else>
                        <template #activator="{ on }">
                            <v-icon v-on="on">check_circle</v-icon>
                        </template>
                        <span>last seen {{ item.last_seen | moment("from", "now") }}</span>
                    </v-tooltip>
                </template>

                <template v-slot:item.uid="{ item }">
                    <v-chip>
                        {{ item.uid }}
                        <v-icon small right @click.stop v-clipboard="item.uid" v-clipboard:success="showCopySnack">mdi-content-copy</v-icon>
                    </v-chip>
                </template>

                <template v-slot:item.name="{ item }">
                    <v-edit-dialog :return-value="editName" large @open="editName = item.name" @save="save(item)">
                        <v-text-field slot="input" v-model="editName" label="Edit" single-line>
                        </v-text-field>
                        <v-icon small left>mdi-file-edit</v-icon>
                        {{ item.name }}
                    </v-edit-dialog>
                </template>

                <template v-slot:item.attributes.pretty_name="{ item }">
                    <v-icon left>{{ deviceIcon[item.attributes.id] || 'fl-tux' }}</v-icon>
                    {{ item.attributes.pretty_name }}
                </template>

                <template v-slot:item.identity.mac="{ item }">
                    <code>{{ item.identity.mac }}</code>
                </template>

                <template v-slot:item.namespace="{ item }">
                  <v-chip class="merda">
                    {{ address(item) }}<v-icon color="white" small right @click.stop v-clipboard="() => address(item)" v-clipboard:success="showCopySnack">mdi-content-copy</v-icon>
                  </v-chip>
                </template>

                <template v-slot:item.actions="{ item }">
                    <v-icon class="icons" @click="featureDevice(item)">
                        info
                    </v-icon>

                    <TerminalDialog :uid="item.uid" v-if="item.online"></TerminalDialog>

                    <v-icon @click="remove(item.uid)">
                        delete
                    </v-icon>
                </template> 
            </v-data-table>
        </v-card-text>
        <v-snackbar v-model="copySnack" :timeout=3000>Device SSHID copied to clipboard</v-snackbar>
    </v-card>
</fragment>
</template>

<script>
import TerminalDialog from "@/components/TerminalDialog.vue";
import AddDevice from "@/components/AddDevice.vue";

export default {
  components: {
    TerminalDialog,
    AddDevice
  },

  created() {
    this.$store.dispatch("devices/fetch");
  },

  computed: {
    devices() {
      return this.$store.getters["devices/list"];
    }
  },

  methods: {
    featureDevice(value){
      this.$router.push('/devices/'+value.uid) 
    },

    address(item) {
      return `${item.namespace}.${item.name}@${this.hostname}`;
    },

    copy(device) {
      this.$clipboard(device.uid);
    },

    remove(uid) {
      if (confirm("Are you sure?")) {
        this.$store.dispatch("devices/remove", uid);
      }
    },

    showCopySnack() {
      this.copySnack = true;
    },

    save(item) {
      this.$store.dispatch("devices/rename", {
        uid: item.uid,
        name: this.editName
      });

      item.name = this.editName;
    }
  },

  data() {
    return {
      hostname: window.location.hostname,
      deviceIcon: {
        arch: "fl-archlinux",
        ubuntu: "fl-ubuntu"
      },
      copySnack: false,
      editName: "",
      headers: [
        {
          text: "Online",
          value: "online",
          align: "center"
        },
        {
          text: "Name",
          value: "name"
        },
        {
          text: "Operating System",
          value: "attributes.pretty_name"
        },

        {
          text: "SSHID",
          value: "namespace",
          align: "center"
        },
        {
          text: "Actions",
          value: "actions",
          align: "center",
          sortable: false
        }
      ]
    };
  }
};
</script>
<style scoped>
.merda {
  font-family: monospace;
}

.icons{
  margin-right: 4px;
}
</style>