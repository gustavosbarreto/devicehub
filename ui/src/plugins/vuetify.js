import Vue from 'vue';
import Vuetify from 'vuetify/lib';
import '@mdi/font/css/materialdesignicons.css';
import 'font-logos/assets/font-logos.css';
import { library } from '@fortawesome/fontawesome-svg-core';
import { FontAwesomeIcon } from '@fortawesome/vue-fontawesome';
import { fas } from '@fortawesome/free-solid-svg-icons';
import Clipboard from 'v-clipboard';
import Fragment from 'vue-fragment';

//import 'vuetify/dist/vuetify.min.css'
//import '../styles/variables.scss'

Vue.component('font-awesome-icon', FontAwesomeIcon); // Register component globally
library.add(fas); // Include needed icons.

Vue.use(Vuetify);
Vue.use(Clipboard);
Vue.use(Fragment.Plugin);

const DEVICE_ICONS = {
  alpine: 'fl-alpine',
  arch: 'fl-archlinux',
  centos: 'fl-centos',
  coreos: 'fl-coreos',
  debian: 'fl-debian',
  devuan: 'fl-devuan',
  elementary: 'fl-elementary',
  fedora: 'fl-fedora',
  freebsd: 'fl-freebsd',
  gentoo: 'fl-gentoo',
  linuxmint: 'fl-linuxmint',
  mageia: 'fl-mageia',
  manjaro: 'fl-manjaro',
  mandriva: 'fl-mandriva',
  nixos: 'fl-nixos',
  opensuse: 'fl-opensuse',
  rhel: 'fl-redhat',
  sabayon: 'fl-sabayon',
  slackware: 'fl-slackware',
  ubuntu: 'fl-ubuntu',
  raspbian: 'fl-raspberry-pi',
  'ubuntu-core': 'fl-ubuntu',
  void: 'fl-void',
};

export default new Vuetify({
  iconfont: 'md',
  icons:{
    values:DEVICE_ICONS
  },
  theme: {
    dark: false,
    themes: {
      light: {
        primary: '#364576',
        secondary: '#b0bec5',
        accent: '#8c9eff',
        error: '#b71c1c',
      },
    }
  }
});
