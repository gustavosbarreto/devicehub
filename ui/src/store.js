import Vue from 'vue';
import Vuex from 'vuex';
import stats from '@/modules/stats';
import sessions from '@/modules/sessions';
import auth from '@/modules/auth';
import devices from '@/modules/devices';
import modals from '@/modules/modals';
import firewallrules from '@/modules/firewall_rules';
import notifications from '@/modules/notifications';

Vue.use(Vuex);

export default new Vuex.Store({
  modules: {
    devices, modals, stats, sessions, auth, firewallrules, notifications,
  },
});
