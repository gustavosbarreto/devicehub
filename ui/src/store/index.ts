import { InjectionKey } from "vue";
import { createStore, Store, useStore as vuexUseStore } from "vuex";

import { auth, AuthState } from "./modules/auth";
import { apiKeys, ApiKeysState } from "./modules/api_keys";
import { layout, LayoutState } from "./modules/layout";
import { users, UsersState } from "./modules/users";
import { tags, TagsState } from "./modules/tags";
import { stats, StatsState } from "./modules/stats";
import { spinner, SpinnerState } from "./modules/spinner";
import { snackbar, SnackbarState } from "./modules/snackbar";
import { sessions, SessionsState } from "./modules/sessions";
import { security, SecurityState } from "./modules/security";
import { publicKeys, PublicKeysState } from "./modules/public_keys";
import { privateKey, PrivateKeyState } from "./modules/private_key";
import { notifications, NotificationsState } from "./modules/notifications";
import { modal, ModalState } from "./modules/modal";
import { firewallRules, FirewallRulesState } from "./modules/firewall_rules";
import { devices, DevicesState } from "./modules/devices";
import { box, BoxState } from "./modules/box";
import { namespaces, NamespacesState } from "./modules/namespaces";
import { billing } from "./modules/billing";
import { customer, CustomerState } from "./modules/customer";
import { announcement, AnnouncementState } from "./modules/announcement";
import apiPlugin from "./plugins/api";

export interface State {
  auth: AuthState;
  apiKeys: ApiKeysState;
  billing: NamespacesState;
  customer: CustomerState;
  box: BoxState;
  devices: DevicesState;
  firewallRules: FirewallRulesState;
  layout: LayoutState;
  modal: ModalState;
  namespaces: NamespacesState;
  notifications: NotificationsState;
  privateKey: PrivateKeyState;
  publicKeys: PublicKeysState;
  security: SecurityState;
  sessions: SessionsState;
  snackbar: SnackbarState;
  spinner: SpinnerState;
  stats: StatsState;
  tags: TagsState;
  users: UsersState;
  announcement: AnnouncementState;
}

export const key: InjectionKey<Store<State>> = Symbol("store");

export const store = createStore<State>({
  modules: {
    auth,
    apiKeys,
    customer,
    billing,
    box,
    devices,
    firewallRules,
    layout,
    modal,
    namespaces,
    notifications,
    privateKey,
    publicKeys,
    security,
    sessions,
    snackbar,
    spinner,
    stats,
    tags,
    users,
    announcement,
  },
  plugins: [
    apiPlugin,
  ],
});

export function useStore(): Store<State> {
  return vuexUseStore(key);
}
