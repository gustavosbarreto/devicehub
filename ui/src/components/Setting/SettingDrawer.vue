<template>
  <v-navigation-drawer
    v-model="showNavigationDrawer"
    :permanent="permanent"
    class="bg-v-theme-surface"
    app
  >
    <v-list
      density="compact"
      :items="visibleItems"
      item-value="path"
      class="pa-0"
    >
      <template #item="{ props }">
        <v-list-item
          :key="props.value"
          :title="props.title"
          :to="props.value"
        />
      </template>
    </v-list>
  </v-navigation-drawer>
</template>

<script setup lang="ts">
import { computed, ref } from "vue";
import { useDisplay } from "vuetify";
import { envVariables } from "@/envVariables";
import { useStore } from "@/store";

const { lgAndUp } = useDisplay();

const permanent = computed(() => lgAndUp.value);
const showNavigationDrawer = ref(true);
const store = useStore();
const namespacedInstance = computed(() => localStorage.getItem("tenant") !== "");
const hasNamespace = computed(() => store.getters["namespaces/getNumberNamespaces"] !== 0);

const items = computed(() => [
  {
    title: "Profile",
    path: "/settings/profile",
  },
  {
    title: "Namespace",
    path: "/settings/namespace",
    hidden: !namespacedInstance.value,
  },
  {
    title: "Private Keys",
    path: "/settings/private-keys",
  },
  {
    title: "Tags",
    path: "/settings/tags",
    hidden: !namespacedInstance.value,
  },
  {
    title: "Billing",
    path: "/settings/billing",
    hidden: !(
      envVariables.billingEnable
          && envVariables.isCloud
          && hasNamespace.value
    ),
  },
]);

const visibleItems = computed(() => items.value.filter((item) => !item.hidden));

defineExpose({ visibleItems });
</script>
