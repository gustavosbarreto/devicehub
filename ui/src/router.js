import Vue from "vue";
import Router from "vue-router";
import Dashboard from "./views/Dashboard.vue";
import store from './store'

Vue.use(Router);

const router = new Router({
    mode: "history",
    base: process.env.BASE_URL,
    routes: [
        {
            path: "/",
            name: "dashboard",
            component: Dashboard
        },
        {
            path: "/devices",
            name: "devices",

            component: () =>
                import(/* webpackChunkName: "devices" */ "./views/Devices.vue")
        },
        {
            path: "/devices/:id",
            name: "featureDevice",
            component: () =>
                import("./views/FeatureDevice.vue") 
        },
        {
            path: "/sessions",
            name: "sessions",
            component: () =>
                import("./views/Sessions.vue")
        },
        {
            path: "/login",
            name: "login",
            component: () =>
                import("./views/Login.vue")
        }
    ]
});

router.beforeEach((to, from, next) => {
    if (to.path !== '/login') {
        if (store.getters['auth/isLoggedIn']) {
          return next()
        }

        return next(`/login?redirect=${to.path}`)
      } else {
        return next()
      }
})

export default router;