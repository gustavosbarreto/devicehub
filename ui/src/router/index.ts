import { RouteRecordRaw, createRouter, createWebHistory } from "vue-router";
import { envVariables } from "../envVariables";
import { store } from "@/store";

const Dashboard = () => import("@/views/Dashboard.vue");
const Devices = () => import("@/views/Devices.vue");
const DeviceList = () => import("@/components/Devices/DeviceList.vue");
const DevicePendingList = () => import("@/components/Devices/DevicePendingList.vue");
const DeviceRejectedList = () => import("@/components/Devices/DeviceRejectedList.vue");
const Containers = () => import("@/views/Containers.vue");
const ContainerList = () => import("@/components/Containers/ContainerList.vue");
const ContainerPendingList = () => import("@/components/Containers/ContainerPendingList.vue");
const ContainerRejectedList = () => import("@/components/Containers/ContainerRejectedList.vue");
const Connectors = () => import("@/views/Connectors.vue");
const detailsConnectors = () => import("@/views/ConnectorDetails.vue");
const DetailsDevice = () => import("@/views/DetailsDevice.vue");
const Sessions = () => import("@/views/Sessions.vue");
const DetailsSessions = () => import("@/views/DetailsSessions.vue");
const FirewallRules = () => import("@/views/FirewallRules.vue");
const PublicKeys = () => import("@/views/PublicKeys.vue");
const Settings = () => import("@/views/Settings.vue");
const SettingProfile = () => import("@/components/Setting/SettingProfile.vue");
const SettingNamespace = () => import("@/components/Setting/SettingNamespace.vue");
const SettingPrivateKeys = () => import("@/components/Setting/SettingPrivateKeys.vue");
const SettingTags = () => import("@/components/Setting/SettingTags.vue");
const SettingBilling = () => import("@/components/Setting/SettingBilling.vue");

export const routes: Array<RouteRecordRaw> = [
  {
    path: "/login",
    name: "login",
    meta: {
      layout: "LoginLayout",
      requiresAuth: false,
    },
    component: () => import("../views/Login.vue"),
  },
  {
    path: "/mfa-login",
    name: "MfaLogin",
    beforeEnter: (to, from, next) => {
      // Check if the user is coming from the login route
      if (from.name === "login") {
        // Allow access to MFA login if the user is coming from the login route
        next();
      } else {
        // Redirect to login if the user is not coming from the login route
        next({ name: "login" });
      }
    },
    meta: {
      layout: "LoginLayout",
      requiresAuth: false,
    },
    component: () => import("../views/MfaLogin.vue"),
  },
  {
    path: "/recover-mfa",
    name: "RecoverMfa",
    beforeEnter: (to, from, next) => {
      if (from.name === "MfaLogin") {
        next();
      } else {
        next({ name: "login" });
      }
    },
    meta: {
      layout: "LoginLayout",
      requiresAuth: false,
    },
    component: () => import("../components/AuthMFA/MfaRecover.vue"),
  },
  {
    path: "/recover-mfa/mail-sucessful",
    name: "RecoverMfaMsg",
    beforeEnter: (to, from, next) => {
      if (from.name === "RecoverMfa") {
        next();
      } else {
        next({ name: "login" });
      }
    },
    meta: {
      layout: "LoginLayout",
      requiresAuth: false,
    },
    component: () => import("../components/AuthMFA/MfaMailRecover.vue"),
  },
  {
    path: "/forgot-pass",
    name: "ForgotPassword",
    meta: {
      layout: "LoginLayout",
      requiresAuth: false,
    },
    component: () => import("../views/ForgotPassword.vue"),
  },
  {
    path: "/validation-account",
    name: "ValidationAccount",
    meta: {
      layout: "LoginLayout",
      requiresAuth: false,
    },
    component: () => import("../views/ValidationAccount.vue"),
  },
  {
    path: "/reset-mfa",
    name: "MfaResetValidation",
    meta: {
      layout: "LoginLayout",
      requiresAuth: false,
    },
    component: () => import("../views/MfaResetValidation.vue"),
  },
  {
    path: "/update-password",
    name: "UpdatePassword",
    meta: {
      layout: "LoginLayout",
      requiresAuth: false,
    },
    component: () => import("../views/UpdatePassword.vue"),
  },
  {
    path: "/sign-up",
    name: "SignUp",
    meta: {
      layout: "LoginLayout",
      requiresAuth: false,
    },
    component: () => import("../views/SignUp.vue"),
  },
  {
    path: "/confirm-account",
    name: "ConfirmAccount",
    meta: {
      layout: "LoginLayout",
      requiresAuth: false,
    },
    component: () => import("../views/ConfirmAccount.vue"),
  },
  {
    path: "/",
    name: "Dashboard",
    component: Dashboard,
  },
  {
    path: "/devices",
    name: "devices",
    component: Devices,
    redirect: {
      name: "listDevices",
    },
    children: [
      {
        path: "",
        name: "listDevices",
        component: DeviceList,
      },
      {
        path: "pending",
        name: "pendingDevices",
        component: DevicePendingList,
      },
      {
        path: "rejected",
        name: "rejectedDevices",
        component: DeviceRejectedList,
      },
    ],
  },
  {
    path: "/containers",
    name: "containers",
    beforeEnter: (to, from, next) => {
      if (!envVariables.isEnterprise && !envVariables.isCloud && envVariables.premiumPaywall) {
        store.commit("users/setShowPaywall", true);
      }
      next();
    },
    component: Containers,
    redirect: {
      name: "listContainers",
    },
    children: [
      {
        path: "",
        name: "listContainers",
        component: ContainerList,
      },
      {
        path: "pending",
        name: "pendingContainers",
        component: ContainerPendingList,
      },
      {
        path: "rejected",
        name: "rejectedContainers",
        component: ContainerRejectedList,
      },
    ],
  },
  {
    path: "/containers/connectors",
    name: "connectors",
    component: Connectors,
    beforeEnter: (to, from, next) => {
      if (!envVariables.isEnterprise && !envVariables.isCloud && envVariables.premiumPaywall) {
        store.commit("users/setShowPaywall", true);
      }
      next();
    },
  },
  {
    path: "/containers/connectors/:id",
    name: "detailsConnectors",
    component: detailsConnectors,
  },
  {
    path: "/device/:id",
    name: "detailsDevice",
    component: DetailsDevice,
  },
  {
    path: "/sessions",
    name: "Sessions",
    component: Sessions,
  },
  {
    path: "/sessions/:id",
    name: "detailsSession",
    component: DetailsSessions,
  },
  {
    path: "/firewall/rules",
    name: "firewalls",
    component: FirewallRules,
    beforeEnter: (to, from, next) => {
      if (!envVariables.isEnterprise && !envVariables.isCloud && envVariables.premiumPaywall) {
        store.commit("users/setShowPaywall", true);
      }
      next();
    },
  },
  {
    path: "/sshkeys/public-keys",
    name: "publicKeys",
    component: PublicKeys,
  },
  {
    path: "/settings",
    name: "settings",
    component: Settings,
    redirect: {
      name: "profileSettings",
    },
    children: [
      {
        path: "profile",
        name: "profileSettings",
        component: SettingProfile,
      },
      {
        path: "namespace-manager",
        name: "namespaceSettings",
        component: SettingNamespace,
      },
      {
        path: "private-keys",
        name: "privateKeysSettings",
        component: SettingPrivateKeys,
      },
      {
        path: "tags",
        name: "tagsSettings",
        component: SettingTags,
      },
      {
        path: "billing",
        name: "billingSettings",
        beforeEnter: (to, from, next) => {
          const enabled = envVariables.billingEnable;
          if (enabled) {
            next();
          } else {
            next("/404");
          }
        },
        component: SettingBilling,
      },
    ],
  },
  {
    path: "/:catchAll(.*)",
    redirect: { name: "NotFound" },
  },
  {
    path: "/404",
    name: "NotFound",
    component: () => import("../views/NotFound.vue"),
  },
];

export const router = createRouter({
  history: createWebHistory(),
  routes,
});

router.beforeEach(async (to, from, next) => {
  const isLoggedIn = store.getters["auth/isLoggedIn"];
  const layout = to.meta.layout || "AppLayout";
  const requiresAuth = to.meta.requiresAuth ?? true;

  await store.dispatch("layout/setLayout", layout);

  // Redirect to the appropriate page based on authentication status
  if (requiresAuth && !isLoggedIn) {
    next({ name: "login" }); // Redirect to login page if authentication is required and user is not logged in
  } else if (to.name === "login" && isLoggedIn) {
    next({ path: "/" }); // Redirect from login page to home if user is already logged in
  } else {
    next(); // Continue with the original navigation
  }
});
