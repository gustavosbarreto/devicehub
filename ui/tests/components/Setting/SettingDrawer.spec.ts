import { ref } from "vue";
import { createVuetify } from "vuetify";
import { mount, VueWrapper } from "@vue/test-utils";
import { afterEach, beforeEach, describe, expect, it } from "vitest";
import { VLayout } from "vuetify/components";
import SettingDrawer from "@/components/Setting/SettingDrawer.vue";
import { store, key } from "@/store";
import { router } from "@/router";

const Component = {
  template: "<v-layout><SettingDrawer v-model='show'/></v-layout>",
};

describe("Setting Drawer", () => {
  let wrapper: VueWrapper<unknown>;

  const vuetify = createVuetify();

  beforeEach(async () => {
    wrapper = mount(Component, {
      global: {
        plugins: [[store, key], vuetify, router],
        config: {
          errorHandler: () => { /* ignore global error handler */ },
        },
        components: {
          "v-layout": VLayout,
          SettingDrawer,
        },
      },
      data() {
        return { show: ref(true) };
      },
    });
  });

  afterEach(() => {
    wrapper.unmount();
  });

  it("Is a Vue instance", () => {
    expect(wrapper.vm).toBeTruthy();
  });

  it("Renders the component", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("Data is defined", () => {
    expect(wrapper.vm.$data).toBeDefined();
  });

  it("renders the v-list with correct items", () => {
    const listItems = wrapper.findAll(".v-list-item");
    const visibleItems = wrapper.findComponent(SettingDrawer)?.vm.visibleItems;

    expect(listItems.length).toBe(visibleItems.length);

    listItems.forEach((item, index) => {
      expect(item.text()).toBe(visibleItems[index].title);
    });
  });

  it("toggles the drawer when showNavigationDrawer changes", async () => {
    const drawer = wrapper.findComponent(SettingDrawer);

    expect(drawer.find(".v-navigation-drawer--active").exists()).toBe(true);

    await wrapper.setData({ show: false });

    expect(drawer.find(".v-navigation-drawer--active").exists()).toBe(false);
  });
});
