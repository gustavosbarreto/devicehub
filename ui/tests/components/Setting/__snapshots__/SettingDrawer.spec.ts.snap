// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Setting Drawer > Renders the component 1`] = `
"<div class=\\"v-layout\\">
  <nav class=\\"v-navigation-drawer v-navigation-drawer--left v-navigation-drawer--temporary v-navigation-drawer--active v-theme--light v-navigation-drawer--mobile bg-v-theme-surface\\" style=\\"left: 0px; z-index: 1004; transform: translateX(0px); position: fixed; transition: none; height: calc(100% - 0px - 0px); top: 0px; bottom: 0px; width: 256px;\\" app=\\"\\">
    <!---->
    <!---->
    <div class=\\"v-navigation-drawer__content\\">
      <div class=\\"v-list v-theme--light bg-transparent v-list--density-compact v-list--one-line pa-0\\" tabindex=\\"0\\" role=\\"listbox\\"><a class=\\"v-list-item v-list-item--link v-theme--light v-list-item--density-compact v-list-item--one-line rounded-0 v-list-item--variant-text\\" tabindex=\\"-2\\" aria-selected=\\"false\\" href=\\"/settings/profile\\"><span class=\\"v-list-item__overlay\\"></span><span class=\\"v-list-item__underlay\\"></span>
          <!---->
          <div class=\\"v-list-item__content\\" data-no-activator=\\"\\">
            <div class=\\"v-list-item-title\\">Profile</div>
            <!---->
            <!---->
          </div>
          <!---->
        </a><a class=\\"v-list-item v-list-item--link v-theme--light v-list-item--density-compact v-list-item--one-line rounded-0 v-list-item--variant-text\\" tabindex=\\"-2\\" aria-selected=\\"false\\" href=\\"/settings/namespace\\"><span class=\\"v-list-item__overlay\\"></span><span class=\\"v-list-item__underlay\\"></span>
          <!---->
          <div class=\\"v-list-item__content\\" data-no-activator=\\"\\">
            <div class=\\"v-list-item-title\\">Namespace</div>
            <!---->
            <!---->
          </div>
          <!---->
        </a><a class=\\"v-list-item v-list-item--link v-theme--light v-list-item--density-compact v-list-item--one-line rounded-0 v-list-item--variant-text\\" tabindex=\\"-2\\" aria-selected=\\"false\\" href=\\"/settings/private-keys\\"><span class=\\"v-list-item__overlay\\"></span><span class=\\"v-list-item__underlay\\"></span>
          <!---->
          <div class=\\"v-list-item__content\\" data-no-activator=\\"\\">
            <div class=\\"v-list-item-title\\">Private Keys</div>
            <!---->
            <!---->
          </div>
          <!---->
        </a><a class=\\"v-list-item v-list-item--link v-theme--light v-list-item--density-compact v-list-item--one-line rounded-0 v-list-item--variant-text\\" tabindex=\\"-2\\" aria-selected=\\"false\\" href=\\"/settings/tags\\"><span class=\\"v-list-item__overlay\\"></span><span class=\\"v-list-item__underlay\\"></span>
          <!---->
          <div class=\\"v-list-item__content\\" data-no-activator=\\"\\">
            <div class=\\"v-list-item-title\\">Tags</div>
            <!---->
            <!---->
          </div>
          <!---->
        </a></div>
    </div>
    <!---->
  </nav>
  <transition-stub name=\\"fade-transition\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
    <div class=\\"v-navigation-drawer__scrim\\" style=\\"z-index: 1003;\\"></div>
  </transition-stub>
</div>"
`;
